### PR TITLE
Simplify project structure & fix compilation issues

### PR DESCRIPTION
**Pull request by damieng**
This PR gets the solution compiling again on VS2017.  Specifically it:

1. Fixes the windows solution so it includes Core
2. Updates the UWP projects to target much newer SDK (min ver unchanged)
3. Updates the Android projects to target SDK26 (v8) as required by Google (min ver unchanged)
4. Re-orders the frameworks in Core - shouldn't be needed but it fixed UWP...
5. Removes the Shared project and replaces it with Auth0Base in core
6. Platform-specific parts are now in the relevant csproj with no more conditional defines
7. Various documentation fixes

### CODE REVIEW

File: src/Auth0.OidcClient.Core/IAuth0Client.cs [hunk 1]
```diff
@@ -1,66 +1,69 @@
...
+    /// underlying <see cref="IdentityModel.OidcClient.OidcClient"/>.
+    /// </summary>
     public interface IAuth0Client
     {
         /// <summary>
         /// Launches a browser to log the user in.
         /// </summary>
         /// <param name="extraParameters">Any extra parameters that need to be passed to the authorization endpoint.</param>
-        /// <returns></returns>
+        /// <returns>A <see cref="LoginResult"/> containing the tokens and claims.</returns>
```
Comment (joshcanhelp) [inline src/Auth0.OidcClient.Core/IAuth0Client.cs]: > containing the tokens and claims

What, exactly, does this provide? If this is an auth code grant, then the auth server response is a code, not tokens. Claims are user attributes within the ID token.

File: src/Auth0.OidcClient.Core/IAuth0Client.cs [hunk 2]
```diff
@@ -1,66 +1,69 @@
...
+        /// <returns>A <see cref="BrowserResultType"/> indicating whether the logout was successful.</returns>
         Task<BrowserResultType> LogoutAsync();
 
         /// <summary>
-        /// Launches a browser to log the user out and clear the Auth0 SSO Cookie
+        /// Launches a browser to log the user out and clear the Auth0 SSO Cookie.
         /// </summary>
-        /// <param name="federated">Indicates whether the user should also be logged out of their identity provider.</param>
-        /// <returns></returns>
+        /// <param name="federated">Wether to log the user out of their federated identity provider.</param>
```
Comment (joshcanhelp) [inline src/Auth0.OidcClient.Core/IAuth0Client.cs]: "Whether"

File: src/Auth0.OidcClient.Core/IAuth0Client.cs [hunk 3]
```diff
@@ -1,66 +1,69 @@
...
         /// <summary>
-        /// Generates an <see cref="IdentityModel.OidcClient.AuthorizeState"/> containing the URL, state, nonce and code challenge which can
+        /// Generates an <see cref="AuthorizeState"/> containing the URL, state, nonce and code challenge which can
         /// be used to redirect the user to the authorization URL, and subsequently process any response by calling
         /// the <see cref="ProcessResponseAsync"/> method.
         /// </summary>
-        /// <param name="extraParameters"></param>
-        /// <returns></returns>
+        /// <param name="extraParameters">Additional parameters to send to the login endpoint.</param>
+        /// <returns>A <see cref="AuthorizeState"/> with necessary URLs, nonce, state and code verifiers.</returns>
```
Comment (joshcanhelp) [inline src/Auth0.OidcClient.Core/IAuth0Client.cs]: > code verifiers

What are those?

File: src/Auth0.OidcClient.Core/IAuth0Client.cs [hunk 4]
```diff
@@ -1,66 +1,69 @@
...
         /// <summary>
-        /// Process the response from the Auth0 redirect URI
+        /// Process the response from the Auth0 redirect URI.
         /// </summary>
         /// <param name="data">The data containing the full redirect URI.</param>
-        /// <param name="state">The <see cref="IdentityModel.OidcClient.AuthorizeState"/> which was generated when the <see cref="PrepareLoginAsync"/>
+        /// <param name="state">The <see cref="AuthorizeState"/> which was generated when the <see cref="PrepareLoginAsync"/>
         /// method was called.</param>
-        /// <returns></returns>
+        /// <returns>A <see cref="LoginResult"/> containing the tokens and claims.</returns>
```
Comment (joshcanhelp) [inline src/Auth0.OidcClient.Core/IAuth0Client.cs]: Same comment as above ...

File: src/Auth0.OidcClient.Core/IAuth0Client.cs (hunk 1 above)
Comment (damieng) [reply src/Auth0.OidcClient.Core/IAuth0Client.cs]: It's actually one of these which has tokens and claims https://github.com/IdentityModel/IdentityModel.OidcClient/blob/master/src/Shared/Results/LoginResult.cs

File: src/Auth0.OidcClient.Core/IAuth0Client.cs (hunk 3 above)
Comment (damieng) [reply src/Auth0.OidcClient.Core/IAuth0Client.cs]: It returns one of these https://github.com/IdentityModel/IdentityModel.OidcClient/blob/master/src/Shared/AuthorizeState.cs

File: src/Auth0.OidcClient.Core/IAuth0Client.cs (hunk 4 above)
Comment (damieng) [reply src/Auth0.OidcClient.Core/IAuth0Client.cs]: https://github.com/IdentityModel/IdentityModel.OidcClient/blob/master/src/Shared/Results/LoginResult.cs

**Pull request merged**